File: 3st/Jaeyoung/수열과 구간 쿼리 1.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries) {
    for( auto& p : queries ) 
        for( int i = p[0] ; i < p[1] + 1 ; i++ ) 
            arr[i] += 1;
    return arr;
}

File: 4st/Hani/programmer/수열과 구간 쿼리1.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[arr.length];
        
        for (int i = 0; i < queries.length; i++) {
            for (int j = queries[i][0]; j <= queries[i][1]; j++) {
                arr[j]++;
            } 
        }
        
        for (int i = 0; i < arr.length; i++) {
            answer[i] = arr[i];
        }
        return answer;
    }
}